### PR TITLE
Support TimeOnly, DateOnly

### PR DESCRIPTION
**Pull request by kerams**
Implements https://github.com/fable-compiler/Fable/issues/2519.

`TimeOnly` is represented by a simple number, just like `TimeSpan`. `DateOnly` is represented by `Date` like `DateTime`.

Remaining work
- [x] Tests that comparison, hashing (dictionary lookup) works for both types
- [x] TimeOnly subtraction operator
- [x] TimeOnly.(Try)Parse
- [x] DateOnly.ToString
- [x] DateOnly.(Try)Parse
- [x] DateOnly.Add(Days|Months|Years)
- [x] DateOnly.ToDateTime

Not implementing
- `DateOnly` has a [constructor](https://docs.microsoft.com/en-us/dotnet/api/system.dateonly.-ctor?view=net-6.0#System_DateOnly__ctor_System_Int32_System_Int32_System_Int32_System_Globalization_Calendar_) which lets you specify the calendar used to intepret the other parameters. This would add another layer of complexity, and I don't think it's of any interest to the vast majority of users.
- DateOnly.ToLongDateString, DateOnly.ToShortDateString, TimeOnly.ToLongTimeString, TimeOnly.ToShortTimeString. You can just use ToString with the corresponding format specifier.
- DateOnly.ToString formats D, r, R, m, M, y, Y because they produce localized text.
- (Try)ParseExact on both types

### CODE REVIEW

File: tests/Main/DateOnlyTests.fs [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+            let dateTime = DateTime (2000, 2, 28, 0, 30, 0, DateTimeKind.Local)
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (2000, 2, 28))
+            let dateTime = DateTime (2000, 2, 28, 0, 30, 0, DateTimeKind.Utc)
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (2000, 2, 28))
+            let dateTime = DateTime.Now
+            DateOnly.FromDateTime dateTime |> equal (DateOnly (DateTime.Now.Year, DateTime.Now.Month, DateTime.Now.Day))
+            //let dateTime = DateTime.MaxValue
+            //DateOnly.FromDateTime dateTime |> equal DateOnly.MaxValue
+            //let dateTime = DateTime.MinValue
+            //DateOnly.FromDateTime dateTime |> equal DateOnly.MinValue
```
Comment (kerams) [inline tests/Main/DateOnlyTests.fs:92]: These tests don't work because DateTime.MinValue/MaxValue is being created with `DateKind.Unspecified` and not UTC, so when you call `getTicks` on it, you get a different number back than what it was created with. Similarly, I also think there is a bug in `create` of `Date.ts` where `setFullYear` is called instead of `setUTCFullYear` when the requested kind is UTC.

**Comment by alfonsogarciacaro** [comment]
Great work @kerams, thank you! Please just let me know when this is ready to merge 👍

File: tests/Main/DateOnlyTests.fs (hunk 1 above)
Comment (kerams) [reply tests/Main/DateOnlyTests.fs:92]: It's not so simple and this scenario is a corner case with little practical use. Let's forget it.

**Comment by kerams** [comment]
For some reason .NET SDK 5.0.403 was used for the last build, hence the failing tests.

**Comment by kerams** [comment]
Ah, you changed the workflow file today. Well, without .NET 6 we're not going to get liftoff here :).

**Comment by alfonsogarciacaro** [comment]
Sorry, we had some issues when updating to F# 6 (see #2605) but hopefully they're solved now, can you try syncing with `main`. Thanks!

**Comment by kerams** [comment]
@alfonsogarciacaro, I'm not sure what's going on. The new tests pass the first time around, but then do not get compiled as .NET 6 following `/home/runner/work/Fable/Fable> node ./src/fable-compiler-js/src/app.fs.js tests/Main/Fable.Tests.fsproj build/tests-js`. I've switched `fable-compiler-js.fsproj` to .NET 6, but it didn't help.

**Comment by alfonsogarciacaro** [comment]
@kerams Sorry, I should have explained this to you before. Fable standalone packs its own BCL DLLs because the point is dotnet/nuget doesn't need to be installed in the system (or the browser). These DLLs are in fable-metadata and are generated by custom build of FCS by (you probably guessed it) @ncave. We haven't updated the DLLs to net6 yet so that explains why fable-standalone doesn't find the new classes 

We can update the DLLs but maybe it's better to do it when @ncave updates fcs-fable... Yet another fork to make FCS compilable by Fable :) So for now we can disable them for fable-compiler-js. I can do that so you don't need to worry.

Thanks again for this work!

**Comment by kerams** [comment]
Alright then, thanks.

**Comment by ncave** [comment]
@alfonsogarciacaro I've updated the BCL to 6.0.100 in #2612, hopefully that works.

**Comment by alfonsogarciacaro** [comment]
Thank you both! Released in v3.6.0-beta-001 👍
